### PR TITLE
node_exporter 0.15.0 compatibiity

### LINKED ISSUE #33: Default Node Exporter Collectors
The node exporter has updated and has decent defaults. We should change the params to simply not specify any collectors as the default.

### LINKED ISSUE #70: node_expoerter v0.15.0
The release 0.15.0 of node collector changes the configuration of the collectors..

> [CHANGE] Replace --collectors.enabled with per-collector flags #640

https://github.com/prometheus/node_exporter/pull/640

this renders all handling as it currently is implemented of the collectors in this module useless

also https://github.com/voxpupuli/puppet-prometheus/issues/33  is propably obsoleted by this

### PR DESCRIPTION
**Pull request by TheMeier**
fixes #70
fixes #33

change collector parameterization to be compatible with node_exporter 0.15.0

this should be released with a major version since it breaks paramter compatinbility

for two step migration of profiles and roles using this keep node_exporter::collectors paramter as noop with deprecation warning for at least one release

### CODE REVIEW

File: manifests/node_exporter.pp [hunk 1]
```diff
@@ -76,30 +85,32 @@
...
-  $package_ensure       = $::prometheus::params::node_exporter_package_ensure,
-  $package_name         = $::prometheus::params::node_exporter_package_name,
-  $purge_config_dir     = true,
-  $restart_on_change    = true,
-  $service_enable       = true,
-  $service_ensure       = 'running',
-  $service_name         = 'node_exporter',
-  $user                 = $::prometheus::params::node_exporter_user,
-  $version              = $::prometheus::params::node_exporter_version,
+  $arch                             = $::prometheus::params::arch,
```
Comment (bastelfreak) [inline manifests/node_exporter.pp]: please add datatypes to those parameters. An example can be found at https://github.com/voxpupuli/puppet-prometheus/pull/68/files

File: manifests/node_exporter.pp [hunk 2]
```diff
@@ -113,19 +124,28 @@
   validate_bool($manage_user)
```
Comment (bastelfreak) [inline manifests/node_exporter.pp]: please drop those validate_* calls. They are not needed if we have datatypes.

File: manifests/node_exporter.pp [hunk 3]
```diff
@@ -113,19 +124,28 @@
...
   validate_bool($manage_service)
   validate_bool($restart_on_change)
-  validate_array($collectors)
+  if $collectors != undef {
```
Comment (bastelfreak) [inline manifests/node_exporter.pp]: this is enough:

```puppet
if $collectors  { }
```

**Comment by bastelfreak** [comment]
Thanks for this PR @TheMeier!
* I added a few inline comments, can you please review them?
* Please check your used email address, it isn't associated with your github account

File: manifests/node_exporter.pp (hunk 1 above)
Comment (TheMeier) [reply manifests/node_exporter.pp]: hm isn't adding datatypes to all params a different aspect that should be adressed in a seperate PR?

well i did it anyway :)

File: manifests/node_exporter.pp (hunk 2 above)
Comment (TheMeier) [reply manifests/node_exporter.pp]: done

File: manifests/node_exporter.pp (hunk 3 above)
Comment (TheMeier) [reply manifests/node_exporter.pp]: done

**Review by juniorsysadmin — APPROVED** [review]

**Comment by bastelfreak** [comment]
Hi @TheMeier, thanks for the updates. Can you check the used email address in your first commit and add it to your github profile? Otherwise the commti isn't associated with you (or change the email address in the commit).

**Comment by TheMeier** [comment]
email should be fine now

**Comment by bastelfreak** [comment]
Thanks!

**Pull request merged**